src/scripts/main.js

```diff
@@ -1,3 +1,178 @@
+      const salaryB = formatSalary(cellB);
+
+      return isAscending ? salaryA - salaryB : salaryB - salaryA;
+    } else {
+      return isAscending
+        ? cellA.localeCompare(cellB)
+        : cellB.localeCompare(cellA);
+    }
+  });
+}
+
+function updateTable(sortedRows) {
+  const tbody = document.querySelector('tbody');
+
+  tbody.innerHTML = '';
+
+  sortedRows.forEach((row) => tbody.appendChild(row));
+}
+
+function handleSortAndUpdate(index, arraySort, directionSort) {
+  const isAscending = directionSort[index];
+
+  const sortedRows = sortRows(index, isAscending, arraySort);
+
+  updateTable(sortedRows);
+
+  directionSort[index] = !isAscending;
+}
+
+tableHeaders.forEach((header, index) => {
+  sortDirection[index] = true;
+
+  header.addEventListener('click', () => {
+    handleSortAndUpdate(index, rowsArray, sortDirection);
+  });
+});
+
+const rows = document.querySelectorAll('tbody tr');
+
+rows.forEach((row) => {
+  row.addEventListener('click', () => {
+    rows.forEach((r) => r.classList.remove('active'));
+
+    row.classList.add('active');
+  });
+});
+
+const formHTML = `
+  <form class="new-employee-form">
+    <label>
+      Name: <input name="name" type="text" data-qa="name" required>
+    </label>
+    <label>
+      Position: <input name="position" type="text" data-qa="position">
+    </label>
+    <label>
+      Age: <input name="age" type="number" data-qa="age" required>
+    </label>
+    <label>
+      Salary: <input name="salary" type="number" data-qa="salary" required>
+    </label>
+    <label>
+      Office:
+      <select name="office" data-qa="office">
+        <option value="Tokyo">Tokyo</option>
+        <option value="Singapore">Singapore</option>
+        <option value="London">London</option>
+        <option value="New York">New York</option>
+        <option value="Edinburgh">Edinburgh</option>
+        <option value="San Francisco">San Francisco</option>
+      </select>
+    </label>
+    <button type="submit">Save to table</button>
+  </form>
+`;
+
+document.body.insertAdjacentHTML('beforeend', formHTML);
+
+const form = document.querySelector('.new-employee-form');
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+  validateForm();
+});
+
+function validateForm() {
+  const nameEmployee = form.elements.name.value;
+  const position = form.elements.position.value;
+  const office = form.elements.office.value;
+  const age = parseInt(form.elements.age.value, 10);
+  const salary = parseFloat(form.elements.salary.value);
+
+  const validationError = validateEmployeeDetails(
+    nameEmployee,
+    position,
+    age,
+    salary,
+  );
+
+  if (validationError) {
+    showNotification('Error', validationError, 'error');
+
+    return;
+  }
+
+  addEmployeeAndResetForm(nameEmployee, position, office, age, salary);
+  showNotification('Success', 'Employee added successfully!', 'success');
+}
+
+function validateEmployeeDetails(nameWorker, position, age, salary) {
```

Comment:
The function `validateEmployeeDetails` is returning error messages. It would be more flexible to return an object with a boolean `isValid` property and an `errors` property which is an array of error messages. This would allow you to show multiple error messages at once.